src/lib/Components/Artist/Header.tsx

```diff
@@ -115,59 +103,92 @@ class Header extends React.Component<Props, State> {
     return leadingSubstring + " " + birthday
   }
 
-  @track((props, state) => ({
-    action_name: state.following ? Schema.ActionNames.ArtistUnfollow : Schema.ActionNames.ArtistFollow,
+  @track(props => ({
+    action_name: props.artist.isFollowed ? Schema.ActionNames.ArtistUnfollow : Schema.ActionNames.ArtistFollow,
     action_type: Schema.ActionTypes.Tap,
     owner_id: props.artist.internalID,
     owner_slug: props.artist.slug,
     owner_type: Schema.OwnerEntityTypes.Artist,
   }))
   handleFollowChange() {
-    const newFollowersCount = this.state.following ? this.state.followersCount - 1 : this.state.followersCount + 1
-    ARTemporaryAPIModule.setFollowArtistStatus(
-      !this.state.following,
-      this.props.artist.internalID,
-      (error, following) => {
-        if (error) {
-          console.warn(error)
-          this.failedFollowChange()
-        } else {
-          this.successfulFollowChange()
-        }
-        this.setState({ following, followersCount: newFollowersCount })
+    const {
+      relay,
+      artist: { slug, id, isFollowed },
+    } = this.props
+
+    this.setState(
+      {
+        isFollowedChanging: true,
+      },
+      () => {
+        commitMutation<HeaderFollowArtistMutation>(relay.environment, {
+          onCompleted: () => this.successfulFollowChange(),
+          mutation: graphql`
+            mutation HeaderFollowArtistMutation($input: FollowArtistInput!) {
+              followArtist(input: $input) {
+                artist {
+                  id
+                  isFollowed
+                }
+              }
+            }
+          `,
+          variables: {
+            input: {
+              artistID: slug,
+              unfollow: isFollowed,
+            },
+          },
+          optimisticResponse: {
+            followArtist: {
+              artist: {
+                id,
+                isFollowed: !isFollowed,
+              },
+            },
+          },
+          updater: store => {
```

Comment:
I don't _think_ you need this updater function since the relay store will update automatically and the shape of the mutation response looks standard.

Reply:
Ok, looks like it's needed on the Gene mutation but not the Artist. Updated!